js/ui.js

```diff
@@ -0,0 +1,385 @@
+		// when the mouse exits the element
+		button.onmouseout = function() {
+			if(play_displayed) {
+				button.setAttribute("src", button_urls.play);
+			}
+			else {
+				button.setAttribute("src", button_urls.pause);
+			}
+		};
+	}
+	
+	// Function to programmatically add DOM elements to the dataset
+	// drop-down menu and the dataset drop-down menu
+	function populate_menus() {
+		var year_menu = document.getElementById("year_menu");
+		var set_menu = document.getElementById("dataset_menu");
+		
+		// Make sure we actually have some datasets
+		if(datasets) {
+		
+			// A little trick to grab the first element in the object
+			for(var key in datasets) break;
+			
+			current_set = key;
+			
+			if(datasets[key].years) {
+				pop_years(key);
+			}
+			
+			for(var key2 in datasets) {
+				var set = document.createElement("option");
+				set.setAttribute("value", datasets[key2].var_name);
+				var text = document.createTextNode(datasets[key2].var_name);
+				set.appendChild(text);
+				dataset_menu.appendChild(set);
+			}
+		}
+		
+		
+		
+		year_menu.onchange = function() {
+			year_change(this.value);
+		};
+		
+		dataset_menu.onchange = function() {
+			dataset_change(this.value);
+		};
+	}
+	
+	// Callback function when the user selects a new year
+	// in the year drop-down menu
+	function year_change(year) {
+		
+		// Invoke the data-controller!
+		maverick.data.request(query_callback, year, current_set, false, false,
+								false, false, false, false, false, false, false);
+	}
+	
+	// Callback function that the data-controller calls when
+	// it is done querying the fusion table
+	function query_callback(data) {
+
+		for(var key in data) {
+			if(data[key] <= 1.0) {
+				data[key] *= 100;
+			}
+		}
+		
+		response_data = data;
+		
+		console.log(data);
+		// Clean the canvas slate
+		context.clearRect(0, 0, canvas.width, canvas.height);
+		draw();
+	}
+	
+	// Callback function when the user selects a new dataset
+	// from the dataset drop-down menu
+	function dataset_change(set) {
+		
+		// First, we find which dataset was selected
+		for(var key in datasets) {
+			if(datasets[key].var_name === set) {
+				var menu = document.getElementById("year_menu");
+				menu.innerHTML = "";
+				
+				pop_years(key);
+				
+				current_set = key;
+				
+				// We have to query the data to draw the graph
+				// for the new dataset
+				year_change(datasets[key].years[0]);
+				
+				// We also need to change the labels on the graph
+				var lbls = datasets[key].labels;
+				//document.getElementsByClassName("first_label")[0].replaceChild(document.createTextNode(lbls[0]));
+				var first_label = document.getElementsByClassName("first_label")[0];
+				first_label.innerHTML = lbls[0];
+				
+				var rest = document.getElementsByClassName("x_label");
+				for(var i = 1; i < lbls.length; i++)
+				{
+					rest[i-1].innerHTML = lbls[i];
+				}
+				break;
+			}
+		}
+	}
+	
+	// Determine the maximum scale value
+	function find_highest() {
```

Comment:
Is there a compelling reason to not set this at some predetermined value (such as 100)?